File: app/src/main/java/ru/gb/lesson12/data/Repo.java
package ru.gb.lesson12.data;

import java.util.ArrayList;
import java.util.List;

public interface Repo {
    // CRUD
    // ------
    // Create
    // Read
    // Update
    // Delete
    int  create(Note note);
    Note read(int id);
    void update(Note note);
    void delete(int id);
    void writePref(List<Note> notes);
    List<Note> readPref(String notesString);
    List<Note> getAll();
    int readCounter(List<Note> notes);
}

File: app/src/main/java/ru/gb/notes/domain/Repo.kt
package ru.gb.notes.domain

interface Repo {
    // CRUD
    // ------
    // Create
    // Read
    // Update
    // Delete
    fun create(note: Note?): Int
    fun read(id: Int): Note?
    fun update(note: Note?)
    fun delete(id: Int)
    fun writePref(notes: List<Note?>?)
    fun readPref(notesString: String?): List<Note?>?
    val all: List<Note?>?
    fun readCounter(notes: List<Note?>?): Int
}